src/pls.jl

```diff
@@ -221,6 +221,17 @@ StatsBase.fitted(m::LinearMixedModel{T}) where {T} = fitted!(Vector{T}(nobs(m)),
 
 StatsBase.predict(m::LinearMixedModel) = fitted(m)
 
+function StatsBase.predict(obj::LinearMixedModel,newDF)
+    # just a hack here, we'll generate a new model to have it do the dataframe manipulation
+    new_obj = lmm(obj.formula,newDF)
+    # println(full(new_obj.trms[1]))
+    ranef_total = 0
+    for grp=1:length(obj.trms)-2
+        ranef_total += full(new_obj.trms[grp])*collect(Base.Iterators.flatten(ranef(obj)[grp]))
+    end
```

Comment:
You really don't want to do `full(new_obj.trms[grp])`, which could easily exhaust the available memory for some models with tens of millions of observations and hundreds of thousands of levels in a grouping factor.  Look at the existing `fitted` function and the use of `unscaledre!` for an alternative.

I am okay with writing a version of this if you prefer.  I make these suggestions assuming that you are interested in learning the low-level details.

The point of all of the use of sparse and blocked matrices and the various types of AbstractTerm is to avoid creating the full matrix representation.